contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -117,6 +117,35 @@ abstract contract VaultHub is AccessControlEnumerableUpgradeable {
         return $.sockets[$.vaultIndex[IHubVault(_vault)]];
     }
 
+    /// @notice Returns all vaults owned by a given address
+    /// @param _owner Address of the owner
+    /// @return An array of vaults owned by the given address
+    function vaultsByOwner(address _owner) external view returns (IHubVault[] memory) {
```

Comment:
I'm not sure about the correctness of this method, but such a method will be very useful

Reply:
Maybe needs  `vaultsIndexesByOwner` and then this method around it.

Reply:
The thing is that Vaults are created without an owner, and then the owner and other roles are added. And it seems to me that storing another array in memory that will be updated frequently may be a bad decision.
If this is not the case, then it makes sense to have an additional owner field in the VaultSocket structure.

Reply:
> Maybe needs `vaultsIndexesByOwner` and then this method around it.

We can't maintain the correctness of such index, because the owner can be changed in the vault, independently.